client/blocks/comment-detail/comment-detail-author.jsx

```diff
@@ -0,0 +1,112 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import Gridicon from 'gridicons';
+import { localize } from 'i18n-calypso';
+import classNames from 'classnames';
+
+export class CommentDetailAuthor extends Component {
+	static propTypes = {
+		author: PropTypes.object,
+		commentDate: PropTypes.string,
+	};
+
+	state = {
+		isExpanded: false,
+	};
+
+	toggleExpanded = () => {
+		this.setState( { isExpanded: ! this.state.isExpanded } );
+	}
+
+	render() {
+		const {
+			author,
+			commentDate,
+			moment,
+			translate,
+		} = this.props;
+		const { isExpanded } = this.state;
+
+		const classes = classNames( 'comment-detail__author', {
```

Comment:
It would be a warning if the component was in an index file in its own folder, e.g. `comment-detail/comment-detail-author/index.jsx`.

In this case, it's considered a "sub-component" of `comment-detail`.